metric/metric_recorder.go

```diff
@@ -0,0 +1,59 @@
+package metric
```

Comment:
the file could be called recorder.go btw. It's clear that it's "metric_recorder" based on the package name "metric".